### PR TITLE
Add is_empty() to ArrayBase

### LINKED ISSUE #353: Add is_empty to ArrayBase
I found myself expecting this function while writing array code. Would this make sense to have? I'm totally willing to make a PR if you can point me to the best place to implement. I was thinking the best way would be to add a method to `Dimension` checking if any of the axes are 0, but maybe there is a better way?

### PR DESCRIPTION
**Pull request by iamed2**
Closes #353

### CODE REVIEW

File: src/impl_methods.rs [hunk 1]
```diff
@@ -59,6 +59,11 @@ impl<A, S, D> ArrayBase<S, D> where S: Data<Elem=A>, D: Dimension
         self.dim.size()
     }
 
+    /// Return whether the array has any elements
+    pub fn is_empty(&self) -> bool {
```
Comment (bluss) [inline src/impl_methods.rs]: Thanks for adding this and tests.

Sorry to nitpick on such a small PR -- but `len` & `len_of` must be next to each other in the docs. So this method should move down a step.

Comment (iamed2) [reply src/impl_methods.rs]: Oh no problem! That makes sense.

**Comment by bluss** [comment]
Thanks!

**Comment by iamed2** [comment]
Thank you!

**Comment by bluss** [comment]
In ndarray 0.10.7

**Pull request merged**